### PR TITLE
[cydtseng] iP

### PR DESCRIPTION
**Pull request by cydtseng**
# Duke Project

> *Chatbots represent a new trend in how people access information, make decisions and communicate.”  — Christie Pitts ( Verizon Ventures)*

The Duke Project is a chatbot based todo list with the following features
- Command-based input
- Adding events, deadlines and todos
- Showing list of tasks to do
- All tasks saved locally

How to run:
1. Download the release [here](https://github.com/cydtseng/ip/releases/tag/A-Jar).
2. Run the jar file.
3. Start adding tasks.
4. Let Duke manage tasks for you! :partying_face:

**Interactivity progress:**

- [x] Text based UI
- [ ] GUI with JavaFX 

---
Entry point `main` of Duke program.
```java
public static void main(String[] args) {
    new Duke().run();
}
```

### CODE REVIEW

File: src/main/java/duke/Storage.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return userTasks;
+    }
+
+    public static void saveTasksToFile(ArrayList<Task> taskList) {
+        try {
+            String toWrite = "";
```
Comment (Jayanth-Balasubramanian) [inline src/main/java/duke/Storage.java:117]: A little pedantic, but it might be better to use a noun for this String, such as tasksToWrite

File: src/test/java/ParserTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class ParserTest {
+    Parser parser = new Parser();
+    @Test
+    public void testParseByeCommand(){
+        Assertions.assertTrue(parser.isByeCommand("bye"));
+        Assertions.assertTrue(parser.isByeCommand("bye "));
+        Assertions.assertTrue(parser.isByeCommand(" bye"));
```
Comment (Jayanth-Balasubramanian) [inline src/test/java/ParserTest.java]: It's great that you're testing these edge cases!

File: src/main/java/duke/Parser.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+ * Parser class is used for making sense of all possible user commands.
+ *
+ * @author Tseng Chen-Yu
+ * @version %I%, %G%
+ * @since 1.0
+ */
+public class Parser {
+    private final Stage stage;
+
+    public enum CmdType {
```
Comment (Jayanth-Balasubramanian) [inline src/main/java/duke/Parser.java:28]: enum constants should be in SCREAMING_SNAKE_CASE

File: src/main/java/duke/Parser.java [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    private boolean isEventCommand(String userInput) {
+        return userInput.split(" ")[0]
+                .equals(CmdType.event.name());
+    }
+
+    public boolean isFindCommand(String userInput) {
+        return userInput.split(" ")[0].equals(CmdType.find.name());
+    }
+
+    public String parseCommandWithResponse(String userInput) {
```
Comment (Jayanth-Balasubramanian) [inline src/main/java/duke/Parser.java]: This method seems way too long. As a suggestion, perhaps you could break your method up into handleByeCommand, handleMarkCommand etc so it's easier to debug.

**Review by Jayanth-Balasubramanian — COMMENTED** [review]
LGTM except some minor nitpicks. Very clean code!
